dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/filter/CspFilter.java

```diff
@@ -76,35 +78,33 @@ protected void doFilterInternal( HttpServletRequest req, HttpServletResponse res
         throws ServletException,
         IOException
     {
+        String url = req.getRequestURL().toString();
+
         if ( !enabled )
         {
+            res.addHeader( "X-Frame-Options", "SAMEORIGIN" );
             chain.doFilter( req, res );
             return;
         }
 
-        if ( isStaticContentInsideApi( req.getRequestURI() ) )
+        if ( LOGIN_PATTERN.matcher( url ).matches() )
         {
-            setFrameAncestorsCspRule( res );
-            chain.doFilter( req, res );
-            return;
-        }
+            String nounce = CodeGenerator.getRandomUrlToken();
```

Comment:
Typo, references will also need to be updated

```suggestion
            String nonce = CodeGenerator.getRandomUrlToken();
```